server/src/main/java/io/crate/planner/statement/CopyPlan.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed to Crate.io GmbH ("Crate") under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.  Crate licenses
+ * this file to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.planner.statement;
+
+import static io.crate.data.SentinelRow.SENTINEL;
+
+import java.util.function.BiConsumer;
+import java.util.stream.Collectors;
+
+import io.crate.data.CollectingRowConsumer;
+import io.crate.data.InMemoryBatchIterator;
+import io.crate.data.Row;
+import io.crate.data.Row1;
+import io.crate.data.RowConsumer;
+import io.crate.metadata.TransactionContext;
+
+
+public class CopyPlan {
+
+    public static void execute(RowConsumer consumer,
+                               TransactionContext txnCtx,
+                               BiConsumer<RowConsumer, TransactionContext> biConsumer,
+                               boolean waitForCompletion) {
+
+        if (waitForCompletion) {
+            biConsumer.accept(consumer, txnCtx);
+        } else {
+            biConsumer.accept(new CollectingRowConsumer<>(Collectors.mapping(Row::materialize, Collectors.toList())), txnCtx);
```

Comment:
similar to https://github.com/crate/crate/pull/12245#discussion_r830114135 now its same consumers but anyway have concern:

biconsumer here is JobLauncher execute and we are passing RowConsumer.
in case of exception JobLauncer [will call](https://github.com/crate/crate/blob/b8f5bfc1ae1d34b0426f2a01217fde663505225b/server/src/main/java/io/crate/execution/engine/JobLauncher.java#L154-L156) `passedRowConsumer.accept(null, throwable);`
and then we immediately call `consumer.accept(InMemoryBatchIterator.of(Row1.ROW_COUNT_UNKNOWN, SENTINEL), null);`
basically, it's 2 consumer.accept calls in case of exception  - not sure if it's a problem. @mfussenegger  wdyt?

Reply:
Thanks for mentioning it again. I am not sure. I thought @seut that made the original review would have an idea.

Reply:
@BaurzhanSakhariev in some tests i made we have the following behavior:
In the case of `wait_for_completion = true`, an exception is returned to the user.
In the case of `wait_for_completion = false`, -1 is returned to the user but the copy operation has failed silently.

Reply:
@disideris @BaurzhanSakhariev 
> In the case of wait_for_completion = true, an exception is returned to the user.
> In the case of wait_for_completion = false, -1 is returned to the user but the copy operation is failing silently.

I think this is the expected behaviour.